Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.cpp

```diff
@@ -171,6 +172,16 @@ namespace ScriptCanvas
 
     void AutoGenRegistryManager::UnregisterRegistry(const char* registryName)
     {
-        m_registries.erase(registryName);
+        auto it = m_registries.find(registryName);
+        if (it != m_registries.end())
+        {
+            auto componentDescriptors = it->second->GetComponentDescriptors();
+            for (auto descriptor : componentDescriptors)
+            {
+                descriptor->ReleaseDescriptor();
```

Comment:
for safety concern you could do a nullptr check before calling ReleaseDescriptor

Reply:
Probably not necessary.  Nothing should be setting a descriptor in the `m_cachedDescriptors` to nullptr, and if it's an address pointing to bad memory, we'd want to crash.